File: 프로그래머스/Lv1/같은 숫자는 싫어.js
// 내 풀이
function solution(arr)
{
    let answer = [arr[0]];
    
    for(let i=1; i<arr.length; i++){
        arr[i-1] !== arr[i] ? answer.push(arr[i]) : 0;
    }
    
    return answer;
}

// 다른 사람 풀이
function solution2(arr)
{
    return arr.filter((val,index) => val != arr[index+1]);
}

console.log(solution([1,1,3,3,0,1,1]));
console.log(solution2([4,4,4,3,3]));

File: 프로그래머스/Lv1/같은 숫자는 싫어.py
# 내 풀이
def solution(arr):
    answer = [arr[0]]
    
    for i in range(1, len(arr)):
        if arr[i-1] != arr[i]:
            answer.append(arr[i])
    
    
    return answer

def solution2(s):
    a = []
    for i in s:
        if a[-1:] == [i]: continue
        a.append(i)
    return a

print(solution([1,1,3,3,0,1,1]))
print(solution2([4,4,4,3,3]))